### PR TITLE
JAMES-2666 dead letter integration tests with real group listener

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

File: server/container/guice/protocols/webadmin-mailbox/src/main/java/org/apache/james/modules/server/MailboxRoutesModule.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 import org.apache.james.webadmin.Routes;
 import org.apache.james.webadmin.jackson.QuotaModule;
 import org.apache.james.webadmin.routes.DomainQuotaRoutes;
+import org.apache.james.webadmin.routes.EventDeadLettersRoutes;
```
Comment (chibenwa) [inline server/container/guice/protocols/webadmin-mailbox/src/main/java/org/apache/james/modules/server/MailboxRoutesModule.java]: [NO_REVIEW] is not needed: let's avoid uneeded dependencies

File: server/container/guice/cassandra-guice/src/test/java/org/apache/james/DockerElasticSearchRegistrableExtension.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+package org.apache.james;
+
+import org.apache.james.backends.es.ElasticSearchConfiguration;
+import org.apache.james.util.Host;
+import org.apache.james.util.docker.SwarmGenericContainer;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import com.google.inject.Module;
+
+public class DockerElasticSearchRegistrableExtension implements GuiceModuleTestExtension {
```
Comment (mbaechler) [inline server/container/guice/cassandra-guice/src/test/java/org/apache/james/DockerElasticSearchRegistrableExtension.java]: could you remove "Registrable" from the name?

File: server/protocols/jmap-integration-testing/rabbitmq-jmap-integration-testing/src/test/java/org/apache/james/jmap/rabbitmq/ReindexingWithEventDeadLettersTest.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    private static final ConditionFactory CALMLY_AWAIT = Awaitility
+        .with().pollInterval(ONE_HUNDRED_MILLISECONDS)
+        .and().pollDelay(ONE_HUNDRED_MILLISECONDS)
+        .atMost(Duration.ONE_MINUTE)
+        .await();
+
+    private static SwarmGenericContainer elasticSearchContainer = new SwarmGenericContainer(Images.ELASTICSEARCH_2)
+        .withExposedPorts(ELASTIC_SEARCH_HTTP_PORT, ELASTIC_SEARCH_PORT);
+
+    private static final JamesServerExtensionBuilder.ServerProvider CONFIGURATION_BUILDER = configuration -> GuiceJamesServer
```
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/rabbitmq-jmap-integration-testing/src/test/java/org/apache/james/jmap/rabbitmq/ReindexingWithEventDeadLettersTest.java]: it's not a configuration builder, it's a server builder

File: server/protocols/jmap-integration-testing/rabbitmq-jmap-integration-testing/src/test/java/org/apache/james/jmap/rabbitmq/ReindexingWithEventDeadLettersTest.java [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+            "    }," +
+            "    \"#0\"" +
+            "  ]" +
+            "]";
+
+        with()
+            .header("Authorization", aliceAccessToken.serialize())
+            .body(requestBody)
+            .post("/jmap")
+        .then()
```
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/rabbitmq-jmap-integration-testing/src/test/java/org/apache/james/jmap/rabbitmq/ReindexingWithEventDeadLettersTest.java]: you should not put assertions here

File: server/protocols/jmap-integration-testing/rabbitmq-jmap-integration-testing/src/test/java/org/apache/james/jmap/rabbitmq/ReindexingWithEventDeadLettersTest.java [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+            .log().ifValidationFails()
+            .statusCode(200)
+            .body(NAME, equalTo("messagesSet"))
+            .body(ARGUMENTS + ".notCreated", aMapWithSize(0))
+            .body(ARGUMENTS + ".created", aMapWithSize(1));
+    }
+
+    private List<String> listElasticSearchFailedEvents() {
+        return webAdminApi.with()
+            .get("/events/deadLetter/groups/" + ELASTICSEARCH_LISTENER_GROUP).prettyPeek()
```
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/rabbitmq-jmap-integration-testing/src/test/java/org/apache/james/jmap/rabbitmq/ReindexingWithEventDeadLettersTest.java]: debug spotted

**Comment by Arsnael** [comment]
test this please

**Comment by Arsnael** [comment]
I approve the changes made by @chibenwa

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
```
[a05e413adf163c6c062161a03cb65d795729530e] [ERROR] Errors: 
[a05e413adf163c6c062161a03cb65d795729530e] [ERROR]   MessageWithAttachmentMapperTest.messagesRetrievedUsingFetchTypeFullShouldHaveAttachmentsEmptyWhenNoAttachment » NoHostAvailable
```
test this please

**Comment by Arsnael** [comment]
```
[a05e413adf163c6c062161a03cb65d795729530e]         ℹ︎ Checking the system...
[a05e413adf163c6c062161a03cb65d795729530e]         ✔ Docker version should be at least 1.6.0
[a05e413adf163c6c062161a03cb65d795729530e] 05:40:03.234 [WARN ] o.a.j.u.d.SwarmGenericContainer - 'DOCKER_CONTAINER' environment variable not found, dockering without affinity
[a05e413adf163c6c062161a03cb65d795729530e] 05:40:03.397 [WARN ] o.t.u.RegistryAuthLocator - Failure when attempting to lookup auth config (dockerImageName: testcontainers/rpl5uvfoetopvowi:latest, configFile: /root/.docker/config.json. Falling back to docker-java default behaviour. Exception message: /root/.docker/config.json (No such file or directory)
```
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please weird

**Comment by chibenwa** [comment]
```

[a05e413adf163c6c062161a03cb65d795729530e] [ERROR] Errors: 
[a05e413adf163c6c062161a03cb65d795729530e] [ERROR]   ManageableMailQueueContract.dequeueShouldDecreaseQueueSize:81 » AlreadyClosed ...
[a05e413adf163c6c062161a03cb65d795729530e] [INFO] 
[a05e413adf163c6c062161a03cb65d795729530e] [ERROR] Tests run: 185, Failures: 0, Errors: 1, Skipped: 1
```

test this please

**Comment by Arsnael** [comment]
```
[a05e413adf163c6c062161a03cb65d795729530e] [ERROR] Errors: 
[a05e413adf163c6c062161a03cb65d795729530e] [ERROR]   RabbitMQMailQueueConfigurationChangeTest.increasingBucketCountShouldAllowBrowsingAllQueueElements » IllegalState
```
test this please

**Comment by Arsnael** [comment]
```
[a05e413adf163c6c062161a03cb65d795729530e] 05:56:09.530 [WARN ] o.a.j.u.d.SwarmGenericContainer - 'DOCKER_CONTAINER' environment variable not found, dockering without affinity
[a05e413adf163c6c062161a03cb65d795729530e] 05:56:09.678 [WARN ] o.t.u.RegistryAuthLocator - Failure when attempting to lookup auth config (dockerImageName: testcontainers/kqgd9kstrnuuxgn8:latest, configFile: /root/.docker/config.json. Falling back to docker-java default behaviour. Exception message: /root/.docker/config.json (No such file or directory)
```
test this please

**Comment by Arsnael** [comment]
```
[a05e413adf163c6c062161a03cb65d795729530e] [ERROR] Error occurred in starting fork, check output in log
[a05e413adf163c6c062161a03cb65d795729530e] [ERROR] Process Exit Code: 100
[a05e413adf163c6c062161a03cb65d795729530e] [ERROR] Crashed tests:
[a05e413adf163c6c062161a03cb65d795729530e] [ERROR] org.apache.james.webadmin.routes.TasksRoutesTest
[a05e413adf163c6c062161a03cb65d795729530e] [ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: ExecutionException The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
```
...
test this please

**Comment by Arsnael** [comment]
```
[a05e413adf163c6c062161a03cb65d795729530e] 13:00:28.290 [WARN ] o.a.j.u.d.SwarmGenericContainer - 'DOCKER_CONTAINER' environment variable not found, dockering without affinity
[a05e413adf163c6c062161a03cb65d795729530e] 13:00:28.451 [WARN ] o.t.u.RegistryAuthLocator - Failure when attempting to lookup auth config (dockerImageName: testcontainers/wihvdcbjdwrjsum1:latest, configFile: /root/.docker/config.json. Falling back to docker-java default behaviour. Exception message: /root/.docker/config.json (No such file or directory)
```
test this please

**Comment by trantienduchn** [comment]
```
[a05e413adf163c6c062161a03cb65d795729530e] [INFO] Apache James :: Server :: Data :: LDAP Implementation FAILURE [  02:00 h]
```
tests violated timeout

File: server/container/guice/cassandra-guice/src/test/java/org/apache/james/DockerElasticSearchExtension.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+import com.google.inject.Module;
+
+public class DockerElasticSearchExtension implements GuiceModuleTestExtension {
+    public static final int ELASTIC_SEARCH_PORT = 9300;
+    public static final int ELASTIC_SEARCH_HTTP_PORT = 9200;
+
+    private final SwarmGenericContainer elasticSearchContainer;
+
+    public DockerElasticSearchExtension(SwarmGenericContainer elasticSearchContainer) {
```
Comment (aduprat) [inline server/container/guice/cassandra-guice/src/test/java/org/apache/james/DockerElasticSearchExtension.java]: May be it's time to rename this _Swarm_ not related class.

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**